src/crypto/PrivateKey.ts

```diff
@@ -172,6 +178,71 @@ export class SignedPrivateKey
   }
 }
 
+export class AccountLinkedPrivateKeyV1
+  implements privateKey.AccountLinkedPrivateKey_V1
+{
+  createdNs: Long // time the key was generated, ns since epoch
+  secp256k1: secp256k1 // eslint-disable-line camelcase
+  publicKey: AccountLinkedPublicKey // caches corresponding PublicKey
+
+  constructor(obj: privateKey.AccountLinkedPrivateKey_V1) {
+    if (!obj.secp256k1) {
+      throw new Error('invalid private key')
+    }
+    secp256k1Check(obj.secp256k1)
+    this.secp256k1 = obj.secp256k1
+    this.createdNs = obj.createdNs
+    if (!obj.publicKey) {
+      throw new Error('missing public key')
+    }
+    this.publicKey = new AccountLinkedPublicKey(obj.publicKey)
```

Comment:
Does it make sense to regenerate the `publicKey` from the private key bytes and check?